hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiConfirmationBaseTest.java

```diff
@@ -0,0 +1,272 @@
+                tenantAware.getCurrentTenant(), controllerId, savedAction.getId());
+
+        mvc.perform(get(CONTROLLER_BASE, tenantAware.getCurrentTenant(), controllerId)
+                        .accept(MediaType.APPLICATION_JSON)).andDo(MockMvcResultPrinter.print()).andExpect(status().isOk())
+                .andExpect(jsonPath("$._links.deploymentBase.href", containsString(expectedDeploymentBaseLink)));
+
+    }
+
+    @Test
+    @Description("Controller sends a denied action state.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 1),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 1),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 1),
+            @Expect(type = ActionCreatedEvent.class, count = 1),
+            @Expect(type = TargetUpdatedEvent.class, count = 1),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 3),
+            @Expect(type = TargetUpdatedEvent.class, count = 1),
+            @Expect(type = TargetPollEvent.class, count = 1),
+            @Expect(type = TenantConfigurationCreatedEvent.class, count = 1)})
+    void sendDeniedActionStateFeedbackTest() throws Exception {
+        enableUserConsentFlow();
+
+        final DistributionSet ds = testdataFactory.createDistributionSet("");
+        Target savedTarget = testdataFactory.createTarget("989");
+        savedTarget = getFirstAssignedTarget(assignDistributionSet(ds.getId(), savedTarget.getControllerId()));
+        String controllerId = savedTarget.getControllerId();
+        final Action savedAction = deploymentManagement.findActiveActionsByTarget(PAGE, controllerId)
+                .getContent().get(0);
+
+        sendConfirmationFeedback(savedTarget, savedAction, DdiConfirmationFeedback.Confirmation.DENIED, 10,
+                "Action denied message.").andExpect(status().isOk());
+
+        //asserts that deployment link is not available
+        mvc.perform(
+                get(CONTROLLER_BASE, tenantAware.getCurrentTenant(), controllerId).accept(MediaType.APPLICATION_JSON))
+                .andDo(MockMvcResultPrinter.print()).andExpect(status().isOk())
+                .andExpect(jsonPath("$._links.deploymentBase.href").doesNotExist());
+    }
+
+    private ResultActions sendConfirmationFeedback(final Target target, final Action action,
+            final DdiConfirmationFeedback.Confirmation confirmation, Integer code, String message) throws Exception {
+
+        if (message == null) {
+            message = RandomStringUtils.randomAlphanumeric(1000);
+        }
+
+        final String feedback = getJsonConfirmationFeedback(confirmation, code, Collections.singletonList(message));
+        return mvc.perform(
+                post(CONFIRMNATION_FEEDBACK, tenantAware.getCurrentTenant(), target.getControllerId(), action.getId())
+                        .content(feedback).contentType(MediaType.APPLICATION_JSON));
+    }
```

Comment:
Please verify if action is still exposed by confirmationBase and is not exposed via deploymentBase, since it's denied.

Reply:
done